src/software/thunderscope/field/world_layer.py

```diff
@@ -431,6 +442,83 @@ def draw_robot_status(self, painter):
                     )
                 )
 
+    def draw_robot_speeds(self, painter, robot_speed_map):
+        """Draw the robot speeds
+
+        :param painter: The painter
+
+        """
+        painter.setPen(pg.mkPen(Colors.SPEED_COLOR, width=LINE_WIDTH))
+
+        for robot in self.cached_world.friendly_team.team_robots:
+            start_x = robot.current_state.global_position.x_meters
+            start_y = robot.current_state.global_position.y_meters
+            painter.drawLine(
+                QtCore.QLine(
+                    start_x * MILLIMETERS_PER_METER,
+                    start_y * MILLIMETERS_PER_METER,
+                    (start_x + robot.current_state.global_velocity.x_component_meters)
+                    * MILLIMETERS_PER_METER,
+                    (start_y + robot.current_state.global_velocity.y_component_meters)
+                    * MILLIMETERS_PER_METER,
+                )
+            )
+
+            speed_str = (
+                "ID "
+                + str(robot.id)
+                + " Speed: {:.2f}".format(
+                    math.sqrt(
+                        robot.current_state.global_velocity.x_component_meters ** 2
+                        + robot.current_state.global_velocity.y_component_meters ** 2
+                    )
+                )
+            )
+
+            robot_speed_map[robot.id].setText(str(speed_str))
+            robot_speed_map[robot.id].setPos(
+                (robot.current_state.global_position.x_meters * MILLIMETERS_PER_METER)
+                + 4 * ROBOT_MAX_RADIUS_MILLIMETERS,
+                (robot.current_state.global_position.y_meters * MILLIMETERS_PER_METER)
+                + 4 * ROBOT_MAX_RADIUS_MILLIMETERS,
+            )
```

Comment:
this looks neat, but unfortunately the text on the screen tears up the graphics in pyqt and it makes it look wonky
i think the line it self is very useful and if we needed to see  more specific velocities we can plot the world msg in the future